From: Shiz <hi@shiz.me>
Updated by Rasmus Thomsen <oss@cogitri.dev> on 28th of July 2019
Updated again by Ariadne Conill <ariadne@dereferenced.org> on 31 October 2020
Date: Thu, 20 Aug 2017 01:52:36 +0200
Last-Updated: Sat, 28 Oct 2017 20:23:00 +0200
Subject: [PATCH] Add Alpine targets

This adds `$arch-zel-linux-musl` targets to Rust to encode our toolchain
and distribution-specific quirks instead of polluting the main musl target of
`$arch-unknown-linux-musl`.

--- /dev/null
+++ rustc-1.59.0-src/compiler/rustc_target/src/spec/aarch64_zel_linux_musl.rs
@@ -0,0 +1,13 @@
+use crate::spec::Target;
+
+pub fn target() -> Target {
+    let mut base = super::aarch64_unknown_linux_musl::target();
+
+    base.llvm_target = "aarch64-zel-linux-musl".to_string();
+    base.options.vendor = "zel".to_string();
+    base.options.crt_static_default = false;
+    base.options.static_position_independent_executables = true;
+
+
+    base
+}
--- /dev/null
+++ rustc-1.59.0-src/compiler/rustc_target/src/spec/armv6_zel_linux_musleabihf.rs
@@ -0,0 +1,13 @@
+use crate::spec::Target;
+
+pub fn target() -> Target {
+    let mut base = super::arm_unknown_linux_musleabihf::target();
+
+    base.llvm_target = "armv6-zel-linux-musleabihf".to_string();
+    base.options.vendor = "zel".to_string();
+    base.options.crt_static_default = false;
+    base.options.static_position_independent_executables = true;
+
+
+    base
+}
--- /dev/null
+++ rustc-1.59.0-src/compiler/rustc_target/src/spec/armv7_zel_linux_musleabihf.rs
@@ -0,0 +1,13 @@
+use crate::spec::Target;
+
+pub fn target() -> Target {
+    let mut base = super::armv7_unknown_linux_musleabihf::target();
+
+    base.llvm_target = "armv7-zel-linux-musleabihf".to_string();
+    base.options.vendor = "zel".to_string();
+    base.options.crt_static_default = false;
+    base.options.static_position_independent_executables = true;
+
+
+    base
+}
--- /dev/null
+++ rustc-1.59.0-src/compiler/rustc_target/src/spec/i586_zel_linux_musl.rs
@@ -0,0 +1,14 @@
+use crate::spec::Target;
+
+pub fn target() -> Target {
+    let mut base = super::i686_unknown_linux_musl::target();
+
+    base.options.cpu = "pentium4".to_string();
+    base.llvm_target = "i586-zel-linux-musl".to_string();
+    base.options.vendor = "zel".to_string();
+    base.options.crt_static_default = false;
+    base.options.static_position_independent_executables = true;
+
+
+    base
+}
--- rustc-1.59.0-src.orig/compiler/rustc_target/src/spec/mod.rs
+++ rustc-1.59.0-src/compiler/rustc_target/src/spec/mod.rs
@@ -490,6 +490,16 @@
 }
 
 supported_targets! {
+    ("i586-zel-linux-musl", i586_zel_linux_musl),
+    ("x86_64-zel-linux-musl", x86_64_zel_linux_musl),
+    ("aarch64-zel-linux-musl", aarch64_zel_linux_musl),
+    ("armv6-zel-linux-musleabihf", armv6_zel_linux_musleabihf),
+    ("armv7-zel-linux-musleabihf", armv7_zel_linux_musleabihf),
+    ("powerpc-zel-linux-musl", powerpc_zel_linux_musl),
+    ("powerpc64-zel-linux-musl", powerpc64_zel_linux_musl),
+    ("powerpc64le-zel-linux-musl", powerpc64le_zel_linux_musl),
+    ("s390x-zel-linux-musl", s390x_zel_linux_musl),
+    ("riscv64-zel-linux-musl", riscv64_zel_linux_musl),
     ("x86_64-unknown-linux-gnu", x86_64_unknown_linux_gnu),
     ("x86_64-unknown-linux-gnux32", x86_64_unknown_linux_gnux32),
     ("i686-unknown-linux-gnu", i686_unknown_linux_gnu),
--- /dev/null
+++ rustc-1.59.0-src/compiler/rustc_target/src/spec/powerpc64_zel_linux_musl.rs
@@ -0,0 +1,13 @@
+use crate::spec::Target;
+
+pub fn target() -> Target {
+    let mut base = super::powerpc64_unknown_linux_musl::target();
+
+    base.llvm_target = "powerpc64-zel-linux-musl".to_string();
+    base.options.vendor = "zel".to_string();
+    base.options.crt_static_default = false;
+    base.options.static_position_independent_executables = true;
+
+
+    base
+}
--- /dev/null
+++ rustc-1.59.0-src/compiler/rustc_target/src/spec/powerpc64le_zel_linux_musl.rs
@@ -0,0 +1,13 @@
+use crate::spec::Target;
+
+pub fn target() -> Target {
+    let mut base = super::powerpc64le_unknown_linux_musl::target();
+
+    base.llvm_target = "powerpc64le-zel-linux-musl".to_string();
+    base.options.vendor = "zel".to_string();
+    base.options.crt_static_default = false;
+    base.options.static_position_independent_executables = true;
+
+
+    base
+}
--- /dev/null
+++ rustc-1.59.0-src/compiler/rustc_target/src/spec/powerpc_zel_linux_musl.rs
@@ -0,0 +1,13 @@
+use crate::spec::Target;
+
+pub fn target() -> Target {
+    let mut base = super::powerpc_unknown_linux_musl::target();
+
+    base.llvm_target = "powerpc-zel-linux-musl".to_string();
+    base.options.vendor = "zel".to_string();
+    base.options.crt_static_default = false;
+    base.options.static_position_independent_executables = true;
+
+
+    base
+}
--- /dev/null
+++ rustc-1.59.0-src/compiler/rustc_target/src/spec/s390x_zel_linux_musl.rs
@@ -0,0 +1,13 @@
+use crate::spec::Target;
+
+pub fn target() -> Target {
+    let mut base = super::s390x_unknown_linux_musl::target();
+
+    base.llvm_target = "s390x-zel-linux-musl".to_string();
+    base.options.vendor = "zel".to_string();
+    base.options.crt_static_default = false;
+    base.options.static_position_independent_executables = true;
+
+
+    base
+}
--- /dev/null
+++ rustc-1.59.0-src/compiler/rustc_target/src/spec/x86_64_zel_linux_musl.rs
@@ -0,0 +1,13 @@
+use crate::spec::Target;
+
+pub fn target() -> Target {
+    let mut base = super::x86_64_unknown_linux_musl::target();
+
+    base.llvm_target = "x86_64-zel-linux-musl".to_string();
+    base.options.vendor = "zel".to_string();
+    base.options.crt_static_default = false;
+    base.options.static_position_independent_executables = true;
+
+
+    base
+}
--- /dev/null
+++ rustc-1.52.1-src/compiler/rustc_target/src/spec/riscv64_zel_linux_musl.rs
@@ -0,0 +1,13 @@
+use crate::spec::Target;
+
+pub fn target() -> Target {
+    let mut base = super::riscv64gc_unknown_linux_musl::target();
+
+    base.llvm_target = "riscv64-zel-linux-musl".to_string();
+    base.options.vendor = "zel".to_string();
+    base.options.crt_static_default = false;
+    base.options.static_position_independent_executables = true;
+
+
+    base
+}
